File: firstSeminar/src/main/java/com/seminar/firstSeminar/controller/HealthCheckController.java
package com.seminar.firstSeminar.controller;

import com.seminar.firstSeminar.common.dto.SuccessResponse;
import com.seminar.firstSeminar.dto.HealthCheckResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/health")
public class HealthCheckController {

    @GetMapping("/v1")
    public Map<String, String> healthCheck() {
        Map<String, String> response = new HashMap<>();
        response.put("status", "OK");
        return response;
    }

    @GetMapping("/v2")
    public ResponseEntity<String> healthCheckV2() {
        return ResponseEntity.ok("OK");
    }


    @GetMapping("/v4")
    public ResponseEntity<Map<String, String>> healthCheckV4() {
        Map<String, String> response = new HashMap<>();
        response.put("status", "OK");
        return ResponseEntity.ok(response);
    }
    
    @GetMapping("/v5")
    public ResponseEntity<HealthCheckResponse> healthCheckV5() {
        return ResponseEntity.ok(new HealthCheckResponse("success"));
    }


    @GetMapping("/v6")
    public SuccessResponse healthCheckV6() {
        return SuccessResponse.success("200", "Success", true);
    }

}

File: firstKotlin/src/main/kotlin/org/sopt/firstKotlin/controller/HealthCheckController.kt
package org.sopt.firstKotlin.controller

import org.sopt.firstKotlin.domain.Person
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/health")
class HealthCheckController {
    @GetMapping("/api/v1")
    fun healthCheckV1() = "OK" // 이렇게 단일 표현식으로도 가능하다

    @GetMapping("/api/v2")
    fun healthCheckV2(): ResponseEntity<String> {
        return ResponseEntity.ok("OK")
    }

    @GetMapping("/api/v3")
    fun healthCheckV3(): ResponseEntity<Map<String, String>> {
        val response = HashMap<String, String>()
        response.put("status", "ok") //mc보이가 이미 = 을 사용했으니 난 Put으로 간다.
        return ResponseEntity.ok(response)
    }
    @GetMapping("/api/v4")
    fun healthCheckV4() : ResponseEntity<Person> {
        return ResponseEntity.ok(Person(lastName = "도", firstName = "소현", age = 23))
        //코틀린에서는 new 연산자를 사용하지 않는구나
    }

    @GetMapping("/api/v5")
    fun healthCheckV5() :HealthCheckResponse {
        return HealthCheckResponse("OK",true, "성공")
    }

    //데이터 클래스 사용 ! 클래스 내부에도 사용 가능하다.
    data class HealthCheckResponse(
        val status: String,
        val success: Boolean,
        val message: String
    )

}